app/api/routers/breeze_buddy/signup/handlers.py

```
@@ -0,0 +1,626 @@
+        merchant = await merchant_accessors.create_merchant(
+            merchant_id=request.merchant_id,
+            name=request.name,
+            description=request.description,
+            is_active=True,
+            reseller_id=_SELF_SIGNUP_RESELLER_ID,
+        )
+    except asyncpg.UniqueViolationError:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail=f"Merchant ID '{request.merchant_id}' is already taken.",
+        )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)